api/provider/facebook.go

```diff
@@ -33,14 +37,22 @@ type facebookUser struct {
 
 // NewFacebookProvider creates a Facebook account provider.
 func NewFacebookProvider(ext conf.OAuthProviderConfiguration) (OAuthProvider, error) {
+	authHost := chooseHost(ext.URL, defaultFacebookAuthBase)
+	tokenHost := chooseHost(ext.URL, defaultFacebookTokenBase)
+	profileURL := chooseHost(ext.URL, defaultFacebookAPIBase) + "/me?fields=email,first_name,last_name,name,picture"
```

Comment:
Not critical for this PR, but at some point it might be nice to cleanup some of these hand-built URLs with the `url` package.